### ISSUE TITLE
Error: Exception: Call to a member function getGID() on null

### ISSUE THREAD

**Issue by lesaff** [nextcloud/groupfolders#633]
I am trying to add a group to a newly-created group folder. This is the error in the log.

Nextcloud version: `17.0.0.9`
Group folders version: `5.0.3`
PHP version: `7.3.10`

Error stack

```
[index] Error: Exception: Call to a member function getGID() on null at <<closure>>

0. /####REDACTED####/cloud/public/lib/private/AppFramework/App.php line 126
   OC\AppFramework\Http\Dispatcher->dispatch(OCA\GroupFolders ... {}, "aclMappingSearch")
1. /####REDACTED####/cloud/public/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   OC\AppFramework\App::main("OCA\\GroupFolde ... r", "aclMappingSearch", OC\AppFramework\ ... {}, {id: "2",_route: ... "})
2. <<closure>>
   OC\AppFramework\Routing\RouteActionHandler->__invoke({id: "2",_route: ... "})
3. /####REDACTED####/cloud/public/lib/private/Route/Router.php line 297
   undefinedundefinedcall_user_func(OC\AppFramework\ ... {}, {id: "2",_route: ... "})
4. /####REDACTED####/cloud/public/lib/base.php line 1000
   OC\Route\Router->match("/apps/groupfolders/folders/2/search")
5. /####REDACTED####/cloud/public/index.php line 42
   OC::handleRequest()
```

**Comment by andreasdeschka** [comment]
I have the same error. 

For me the reason was that I also use the following app:
Group SQL Backends https://apps.nextcloud.com/apps/user_sql

In our user external user database I changed the ID of a group, to which I added a rule in the group folders app.

To fix it, I had to go to the nexcloud database and set the ID of the group correctly in this table:
oc_group_folders_groups

The failure only appeared after I deleted the caches in Group SQL Backends.

Maybe t

**Comment by ChristophWurst** [comment]
It happens when a group is removed. see more info in https://github.com/nextcloud/groupfolders/issues/1096

**Comment by jngrb** [comment]
I propose to fix this issue as implemented in my pull request #1395 . Besides this backend issue, my PR also fixes an issue interpreting the data in the frontend (src/components/SharingSidebarView.vue).

**Comment by jngrb** [comment]
@juliushaertl , this issue can be closed, I think. The PR is merged and the duplicate #1096 is also already closed.
